feat(catalogue): federated data access

erikzwart opened the pull request:
### What is this about?

Do **NOT** merge, replaced by #5011

For the DataSHIELD servers Armadillo and Opal we would like to have a status shown on the data-catalogue (catalogue model), this would inform users there is Federated data access available and if the server is up/down.
![image](https://github.com/user-attachments/assets/e952c600-1574-477c-9281-a955ef4e25d6)

### What are the main changes you did
Linked issue: https://github.com/molgenis/GCC/issues/531
- Added a table `Federated data access` to the catalogue model in the `data/_shared` folder
- ~~Added a staging area `Federated data access` in the `data/specific` folder~~
- Added _profiles FederatedDataAccess.yaml
- Bump Catalogue model version to `5.7`

### How to test
- go to the [preview](https://preview-emx2-pr-4795.dev.molgenis.org/apps/central/#/) server
- Log in as admin
- Create a new schema `Federated data access` with the template 'DATA_CATALOGUE_FEDERATED_DATA_ACCESS' and include demo data
- See if the template loads without errors
- Open the schema and check the table `Federated data access`
- Open the schema `Catalogue` and see if the table `Federated data access` exists
- Test if you can store a new `Federated data access` row/item

![image](https://github.com/user-attachments/assets/630236dc-edbc-4a08-bdc1-3564c5e493ed)

### Checklist
- [x] updated docs in case of new feature
- [x] added/updated tests
- [na] added/updated testplan to include a test for this fix, including ref to bug using # notation

erikzwart committed:
Added Federated Data Access table to Catalogue model

erikzwart committed:
New Federated Data Access staging model

erikzwart committed:
bump catalogue model version

BrendaHijmans commented on data/_models/shared/Federated data access.csv:
i am missing profiles here. DataCatalogueFlat for incoporation into the catalogue schema. And if you want a separate staging area for all the server info of all resources, a profile for that, e.g. FederatedDataAccess. Also, in the catalogue model we use composite keys with 'resource' ref to the Resources table and and id. Maybe it is needed/handy to follow to make the queries for the Resources page in UI consistent.

erikzwart replied:
Added profiles DataCatalogueFlat and new profile FederatedDataAccess. Included basic demo dataset.

BrendaHijmans commented on data/_models/specific/Federated data access.csv:
why a duplication in 'specific'?

erikzwart replied:
removed specific

mswertz commented on data/_models/shared/Federated data access.csv:
```diff
@@ -0,0 +1,8 @@
+tableName,tableExtends,tableType,columnName,columnType,key,required,refSchema,refTable,refLink,refBack,refLabel,defaultValue,validation,visible,computed,semantics,profiles,label,description
+Federated data access,,,,,,,,,,,,,,,,,,,List of Armadillo and/or Opal server(s) available for this resource.
+Federated data access,,,id,auto_id,1,true,,,,,,,,,,,,,
```
I would oppose auto id. Please follow same structure as for all other nested resource properties analogous to subpopulations, datasets, etc. So using composite key (resource,name)

erikzwart replied:
Removed auto id, using composite key (resource, name) instead.

erikzwart committed:
removed auto_id, added profiles

erikzwart committed:
delete specific Federated Data Access

erikzwart committed:
Federated Data Access profile

erikzwart committed:
demo data Federated Data Access

erikzwart committed:
Add profile to DataModels.java

erikzwart committed:
sorted Profile list

erikzwart committed:
Add template option SchemaCreateModal.vue

erikzwart committed:
sort templates schemaCreateModel.vue

erikzwart committed:
Profile FederatedDataAccess loader test

erikzwart committed:
fix TestLoaders.java

erikzwart committed:
Fix refTable and add url Federated Data Access model

erikzwart committed:
Merge branch 'master' into feat/catalogue-federated-data-access

erikzwart committed:
update demo data

erikzwart committed:
update profile

erikzwart committed:
fix TestLoaders

erikzwart committed:
Merge branch 'master' into feat/catalogue-federated-data-access

erikzwart committed:
Merge branch 'master' into feat/catalogue-federated-data-access

erikzwart committed:
update model

erikzwart committed:
Merge branch 'master' into feat/catalogue-federated-data-access

erikzwart committed:
disable test

erikzwart committed:
Merge branch 'master' into feat/catalogue-federated-data-access

erikzwart committed:
remove refTable catalogue from Federated data access

erikzwart committed:
reanable test

erikzwart committed:
profile import demo data true

erikzwart committed:
Add profile to Resources

erikzwart committed:
add profile to Version.csv

erikzwart committed:
update profile

erikzwart committed:
Merge branch 'master' into feat/catalogue-federated-data-access

erikzwart committed:
fix typo Federated data access

erikzwart committed:
remove refSchema

erikzwart committed:
demo data Resourses

erikzwart committed:
Added Agent, update demo data

erikzwart committed:
Added Profiles, update demo data and added version change to docs

erikzwart committed:
Update TestLoaders.java

BrendaHijmans commented on data/_models/shared/Federated data access.csv:
```diff
@@ -0,0 +1,11 @@
+tableName,tableExtends,tableType,columnName,columnType,key,required,refSchema,refTable,refLink,refBack,refLabel,defaultValue,validation,visible,computed,semantics,profiles,label,description
+Federated data access,,,,,,,,,,,,,,,,,"DataCatalogueFlat,FederatedDataAccess",,List of Armadillo and/or Opal server(s) available for this resource.
+Federated data access,,,resource,ref,1,TRUE,,Resources,,,,,,,,,"DataCatalogueFlat,FederatedDataAccess",,Federated data access defined for this resource
+Federated data access,,,name,ref,1,TRUE,,Agent,,,,,,,,,"DataCatalogueFlat,FederatedDataAccess",,For example: Armadillo or Opal.
```
Can you make a separate ontology for this and not use Agent? Something like 'DataSHIELD servers'. Agent will be merged with Organisations, and Opal and Armadillo are definitely not organisations.

erikzwart replied:
Reference to Agent removed, added ontology lists.

BrendaHijmans commented on data/_models/shared/Resources.csv:
You do not need the Resources table in your Federated access staging area. You can refer to Resources from your federated access table and use refSchema 'catalogue' refTable 'Resources.

BrendaHijmans commented on data/_demodata/applications/datacatalogue_federated_data_access/Agent.csv:
please change to an ontology table 'DataSHIELD servers', see below

BrendaHijmans commented on data/_models/shared/Federated data access.csv:
```diff
@@ -0,0 +1,11 @@
+tableName,tableExtends,tableType,columnName,columnType,key,required,refSchema,refTable,refLink,refBack,refLabel,defaultValue,validation,visible,computed,semantics,profiles,label,description
+Federated data access,,,,,,,,,,,,,,,,,"DataCatalogueFlat,FederatedDataAccess",,List of Armadillo and/or Opal server(s) available for this resource.
+Federated data access,,,resource,ref,1,TRUE,,Resources,,,,,,,,,"DataCatalogueFlat,FederatedDataAccess",,Federated data access defined for this resource
```
use refSchema 'catalogue' and refTable 'Resources'. Do not import and fill out Resources table in Federated access staging area.

erikzwart committed:
Create ontology DataSHIELD server

erikzwart committed:
Update Federated data access

erikzwart committed:
Update testLoaders.java

erikzwart committed:
Udate Federated data access

erikzwart committed:
Delete DataSHIELD server

erikzwart committed:
Create DataSHIELD servers

erikzwart committed:
Update Federated data access
Create ontology DataSHIELD profile names
Create ontology DataSHIELD profile images

erikzwart committed:
Update DataSHIELD profile images

erikzwart committed:
Update DataSHIELD profile links and descriptions

erikzwart committed:
Add HTTP status codes ontology

erikzwart committed:
Update federated data access status field type

erikzwart committed:
Update HTTP status codes

erikzwart committed:
Update demo data

erikzwart committed:
Update demo data Federated Data Access

erikzwart committed:
Update Federated Data Access staging

erikzwart committed:
Merge branch 'master' into feat/catalogue-federated-data-access

erikzwart committed:
Fix formatting in Agent.csv table

erikzwart committed:
Update Agent

erikzwart committed:
merge master

erikzwart committed:
Merge branch 'master' into feat/catalogue-federated-data-access

erikzwart committed:
Merge branch 'master' into feat/catalogue-federated-data-access

erikzwart committed:
Disable directory and patient registry demo in order to get the previews up.

erikzwart committed:
Update RunMolgenisEmx2

erikzwart committed:
Update RunMolgenisEmx2

erikzwart committed:
Update TestLoaders

erikzwart committed:
Disable testBashScript in TestScriptTask.java

erikzwart committed:
Enable testBashScript
Disable testPython (TestScriptTask)

erikzwart committed:
Format code and remove unused RDF imports

erikzwart committed:
Update TestScriptTask.java

erikzwart committed:
Merge branch 'master' into feat/catalogue-federated-data-access

erikzwart committed:
Update RunMolgenisEmx2.java

erikzwart committed:
Merge branch 'master' into feat/catalogue-federated-data-access

erikzwart commented:
Closing this pr, it's replaced by #5011
